homeassistant/components/flux_led/__init__.py

```diff
@@ -50,43 +50,42 @@ def async_wifi_bulb_for_host(host: str) -> AIOWifiLedBulb:
 
 
 @callback
-def async_name_from_discovery(device: dict[str, Any]) -> str:
+def async_name_from_discovery(device: FluxLEDDiscovery) -> str:
     """Convert a flux_led discovery to a human readable name."""
-    if (mac := device.get(ATTR_ID)) is None:
-        return cast(str, device[ATTR_IPADDR])
-    short_mac = mac[-6:]
-    if device.get(ATTR_MODEL_DESCRIPTION):
-        return f"{device[ATTR_MODEL_DESCRIPTION]} {short_mac}"
-    return f"{device[ATTR_MODEL]} {short_mac}"
+    if device["id"] is None:
+        return device["ipaddr"]
+    short_mac = device["id"][-6:]
+    if device["model_description"]:
+        return f"{device['model_description']} {short_mac}"
+    return f"{device['model']} {short_mac}"
```

Comment:
This is now a typed dict so it wants literal strings

Reply:
You can still use those, when marked as final...

Reply:
Replaced with https://github.com/home-assistant/core/pull/60800 since I couldn't get the CI to pass due to the environment restore issue

Reply:
.. and they are Final now so I reverted the change to the literal string